src/Adapter/File/Uploader/AttachmentFileUploader.php

```diff
@@ -72,31 +70,38 @@ public function __construct(
      * @throws AttachmentNotFoundException
      * @throws AttachmentUploadFailedException
      */
-    public function upload(string $filePath, string $uniqueFileName, int $fileSize, int $id = null): void
-    {
+    public function upload(
+        string $filePath,
+        string $uniqueFileName,
+        int $fileSize,
+        int $id = null,
+        $throwExceptionOnFailure = true
+    ): void {
         $this->checkFileAllowedForUpload($fileSize);
         $this->uploadFile($filePath, $uniqueFileName, $fileSize);
         if ($id !== null) {
-            $this->deleteOldFile($id);
+            $this->deleteOldFile($id, $throwExceptionOnFailure);
         }
     }
 
     /**
-     * @param int $attachmentId
      * @param bool $throwExceptionOnFailure
      *
      * @throws AttachmentNotFoundException
+     * @throws CannotUnlinkAttachmentException
      */
-    private function deleteOldFile(int $attachmentId, $throwExceptionOnFailure = true): void
+    private function deleteOldFile(int $attachmentId, $throwExceptionOnFailure): void
     {
         try {
             $attachment = new Attachment($attachmentId);
             $fileLink = _PS_DOWNLOAD_DIR_ . $attachment->file;
 
             if ($throwExceptionOnFailure) {
-                unlink($fileLink);
-
-                return;
+                try {
```

Comment:
It would make more sense (and it would avoid "two" calls to unlink)to manage the exception throwing in the catch condition:
```php
try {
    unlink($fileLink);
} catch (ErrorException $e) {
    if ($throwExceptionOnFailure) {
        throw new CannotUnlinkAttachmentException($e->getMessage(), 0, null, $fileLink);
    }
}
```